File: python/Best_Time_to_Buy_and_Sell_Stock.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

import sys

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        min_price = sys.maxsize
        
        for cur_price in prices:
            if cur_price < min_price:
                min_price = cur_price
            elif cur_price - min_price > profit:
                profit = cur_price - min_price

        return profit

File: java/leetcode/BestTimeToBuyAndSellStock.java
package leetcode;

//문제 링크: https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

class Solution {
    public int maxProfit(int[] prices) {
        int lowest = Integer.MAX_VALUE;
        int profit = 0;
        for (int p : prices) {
            lowest = Math.min(lowest, p);
            profit = Math.max(profit, p - lowest);
        }
        return profit;
    }
}